cli_tools/common/utils/storage/storage_client.go

```diff
@@ -244,6 +253,18 @@ func SplitGCSPath(p string) (string, string, error) {
 	return "", "", daisy.Errf("%q is not a valid Cloud Storage path", p)
 }
 
+// ConcatGCSPath concatenates multiple elements of GCS path into a GCS path.
```

Comment:
I think I did it this way because `path.Join` removes duplicate slashes, so `path.Join("gs://bucket/", "folder") would result in `gs:/bucket/folder`. Since the first element in the input is provided by customers and already in GCS format (e.g. `gs://bucket`) this would require parsing the input path, extracting bucket/folder, then concatenating it back and appending file name.

Reply:
`!strings.HasSuffix(pathElement, "/")` also removes duplicate slashes, right? I looked at the current usage of `ConcatGCSPath`, and I *think* you can drop-in path.join instead.

Reply:
This func is design to concatenate e.g. "gs://bucket/" with "directory". Since the double slashes are in the middle of an element, they don't get squashed to a single slash.